council/llm/llm_function.py

```diff
@@ -36,37 +39,51 @@ def response(self) -> T_Response:
     @property
     def duration(self) -> float:
         """
-        Get the duration of the LLM's response.
+        Get the duration of the LLM function response.
 
         Returns:
-            float: The time taken by the LLM to produce the response, in seconds.
+            float: The time taken by the LLM function to produce the response, in seconds.
+            This includes total duration of LLM function execution, including self-corrections if any.
         """
-        return self.llm_response.duration
+        return self.llm_response.duration + sum(response.duration for response in self._previous_responses)
 
     @property
     def consumptions(self) -> Sequence[Consumption]:
         """
-        Get the consumptions associated with the LLM's response.
+        Get the consumptions associated with the LLM function response.
 
         Returns:
             Sequence[Consumption]: A sequence of consumption objects if available; otherwise, an empty sequence.
         """
-        result = self.llm_response.result
-        return result.consumptions if result else []
+        consumptions: List[Consumption] = []
+
+        # consumptions of previous self-corrected responses if any
+        for response in self._previous_responses:
+            if response.result is not None:
+                consumptions.extend(response.result.consumptions)
+
+        # consumptions of the final response
+        if self.llm_response.result is not None:
+            consumptions.extend(self.llm_response.result.consumptions)
```

Comment:
Don't we have to sum instead of to list them all? 
Should be consistent w/ duration.

Reply:
I thought about that but decided to go with that approach

In this case consumption is essentially a metric that will be aggregated on the receiver side anyway, and I would say it's better to have asses to raw info